content/systems/filecoin_vm/message/_index.md

```diff
@@ -10,41 +10,139 @@ dashboardTests: 0
+in state. When referring to a Message in the context of Filecoin and unless otherwise defined, we refer to a _Transaction Message_. A message combines:
 
 - a token amount to be transferred from the sender to the receiver, and
-- a method with parameters to be invoked on the receiver (optional).
+- a method with parameters to be invoked on the receiver (optional/where applicable).
 
 Actor code may send additional messages to other actors while processing a received message. 
-Messages are processed synchronously: an actor waits for a sent message to complete before resuming control.
+Messages are processed synchronously, that is, an actor waits for a sent message to complete before resuming control.
 
-The processing of a message consumes units of computation and storage denominated in gas. 
-A message's gas limit provides an upper bound on its computation. The sender of a message pays 
+The processing of a message consumes units of computation and storage, both of which are denominated in gas. 
+A message's _gas limit_ provides an upper bound on the computation required to process it. The sender of a message pays 
 for the gas units consumed by a message's execution (including all nested messages) at a
 gas price they determine. A block producer chooses which messages to include in a block and is
 rewarded according to each message's gas price and consumption, forming a market.
 
 ## Message syntax validation
 
 A syntactically invalid message must not be transmitted, retained in a message pool, or
-included in a block.
+included in a block. Even if an invalid message ends up being sent (e.g., by a malicious player), the libp2p GossipSub propagation protocol will reject the message on the receiving end.
+
+When transmitted individually (before inclusion in a block), a message is packaged as
+`SignedMessage`, regardless of signature scheme used.
+
+```go
+type SignedMessage struct {
+	Message   Message
+	Signature crypto.Signature
+}
+```
 
 A syntactically valid `UnsignedMessage`:
 
 - has a well-formed, non-empty `To` address,
 - has a well-formed, non-empty `From` address, 
-- has a non-negative `CallSeqNum`,
 - has `Value` no less than zero and no greater than the total token supply (`2e9 * 1e18`), and
-- has a non-negative `MethodNum`,
+- has a non-negative `MethodNum`, or zero if the message is for a plain transfer of funds between two actors,
 - has non-empty `Params` only if `MethodNum` is zero,
 - has non-negative `GasPrice`,
 - has `GasLimit` that is at least equal to the gas consumption associated with the message's serialized bytes,
 - has `GasLimit` that is no greater than the block gas limit network parameter.
 
-When transmitted individually (before inclusion in a block), a message is packaged as
-`SignedMessage`, regardless of signature scheme used. A valid signed message:
 
-- has a total serialized size no greater than `message.MessageMaxSize`.
+```go
+type Message struct {
+	// Version of this message
+	Version int64
```

Comment:
I'm surprised this isn't a `uint64` - could you check? If it's not, there should probably be a validation rule about it being non-negative.

Reply:
Ok, I've asked what's up with this [in Slack](https://filecoinproject.slack.com/archives/CP50PPW2X/p1597812900247300).

Reply:
FYI I'm gunna change this to uint64 in lotus